app/controllers/notes_controller.rb

```diff
@@ -3,13 +3,13 @@
 class NotesController < ApplicationController
   def create
     @need = Need.find(params[:need_id])
-    @note = @need.notes.create(note_params.merge(user: current_user))
+    @note = @need.notes.create!(note_params.merge(user: current_user))
     redirect_to need_path(@need)
   end
 
   private
 
   def note_params
-    params.require(:note).permit(:body)
+    params.require(:note).permit(:body, :category)
```

Comment:
Per slack convo, it probably makes sense to require _either_ of these fields, rather than both. So you can submit with category or with body or with category and body but not without either. But I wouldn't call it a blocker necessarily; requiring both is better than not requiring either.

Reply:
Tim, only :note is required, :body and :category are permitted if present.

Reply:
Okay 🤔 I don't get how `note` is a field on a `Note`, on the same level as `body` and `category`, but maybe I'm missing something. I'm sure it's fine.

Reply:
it's fine :-)